### PR TITLE
docs: clarify macOS support of the `inproc` backend

### PR DESCRIPTION
**Pull request by supervacuus**
as a review result of: https://github.com/getsentry/sentry-docs/pull/11721

#skip-changelog

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,6 +270,7 @@ Legend:
 - ☑ default
 - ✓ supported
 - unsupported
+- `*`: `inproc` has not produced valid stack traces on macOS since version 13 ("Ventura"). Details: https://github.com/getsentry/sentry-native/issues/906
```
Comment (kahest) [inline README.md:273]: should we instead/additionally point to the advanced usage page?

**Review by kahest — APPROVED** [review]
thanks for adding, left a question/suggestion

File: README.md [hunk 2]
```diff
@@ -260,7 +260,7 @@ using `cmake -D BUILD_SHARED_LIBS=OFF ..`.
 | - none     | ✓       | ✓     | ✓     | ☑       | ☑   |
 |            |         |       |       |         |     |
 | Backends   |         |       |       |         |     |
-| - inproc   | ✓       | ✓     | ✓     | ☑       |     |
+| - inproc   | ✓       | (✓)*  | ✓     | ☑       |     |
```
Comment (JoshuaMoelans) [inline README.md:263]: nit: is there a reason why we keep `inproc` as the first listed item if macOS doesn't really support it? Maybe we should move the default `crashpad` to the top of this list; as a user that would be less confusing imo

**Review by JoshuaMoelans — APPROVED** [review]
LGTM (+ small nit)

**Review by kahest — APPROVED** [review]
🚀

**Pull request merged**